datacollectiondemo/source/c/CMakeLists.txt

```diff
@@ -15,22 +15,41 @@
 #
 
 cmake_minimum_required(VERSION 2.8.8)
-project (C-SDK-sample C)
 
+project(C-SDK-sample)
+enable_language(C)
+
+# Our application name.
+set(APP_NAME demo_client)
+
+# Target-independent flags.
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c99 -g -Wall -Wextra")
+
+# Path to the Kaa SDK.
+add_subdirectory(libs/kaa)
+# Directory containing target support library.
+add_subdirectory(targets/${KAA_TARGET})
+
+# This is required for ESP8266 platform
+# due to it's specific requirements regarding linked executable.
+# The blank.c file is a placeholder for CMake's add_executable()
+# All the code (Kaa SDK, ESP8266 SDK and demo) is compiled as static libraries
+# and linked into that executable.
+if("${KAA_TARGET}" STREQUAL "esp8266")
+    add_library(${APP_NAME}_s STATIC src/kaa_demo.c)
+    file(WRITE ${CMAKE_BINARY_DIR}/blank.c "")
+    add_executable(${APP_NAME} ${CMAKE_BINARY_DIR}/blank.c)
+    target_link_libraries(${APP_NAME} ${APP_NAME}_s)
+    target_link_libraries(${APP_NAME}_s target_support kaac)
+else()
+    add_executable(${APP_NAME} src/kaa_demo.c)
+    target_link_libraries(${APP_NAME} kaac target_support)
+endif()
+
+# For most embedded targets binary stripping is required.
+if (${KAA_PRODUCE_BINARY})
+    add_custom_command(TARGET ${APP_NAME} POST_BUILD
+            COMMAND ${CMAKE_OBJCOPY} -O binary ${APP_NAME} ${APP_NAME}.bin)
+endif ()
 
-set(APP_NAME "demo_client")
-set(KAA_SDK_DIR "client-c")
-
-execute_process(COMMAND ${CMAKE_COMMAND} -E make_directory ${KAA_SDK_DIR})
-execute_process(COMMAND sh -c "tar -zxf ${CMAKE_CURRENT_SOURCE_DIR}/libs/kaa/kaa-c*.tar.gz -C ${KAA_SDK_DIR}")
```

Comment:
This probably should be left with some modifications.
Let's set `KAA_SDK_DIR` to `${CMAKE_CURRENT_LIST_DIR}/libs/kaa`. Additional check whether SDK has already been unpacked is also preferred.